src/connector/src/parser/mod.rs

```diff
@@ -229,10 +229,9 @@ impl<'a> MessageMeta<'a> {
                     KAFKA_TIMESTAMP_COLUMN_NAME,
                     "unexpected kafka meta column name"
                 );
-                kafka_meta.timestamp.map(|ts| {
-                    risingwave_common::cast::i64_to_timestamptz(ts)
-                        .unwrap()
-                        .into()
+                kafka_meta.timestamp.to_millis().and_then(|ts_millis| {
+                    risingwave_common::types::Timestamptz::from_millis(ts_millis)
+                        .map(ScalarRefImpl::Timestamptz)
```

Comment:
DONT REPEAT YOURSELF
```suggestion
                kafka_meta.extract_timestamp()
```